app/src/main/java/com/example/xpvehicles/activities/VehicleDetailsActivity.java

```diff
@@ -131,64 +138,148 @@ private void setValues() {
+        materialDateBuilder.setCalendarConstraints(calendarConstraintBuilder.build());
+        MaterialDatePicker<Long> materialDatePicker = materialDateBuilder.build();
+        materialDatePicker.show(getSupportFragmentManager(), "MATERIAL_DATE_PICKER");
+
+        // positive button == ok button
+        materialDatePicker.addOnPositiveButtonClickListener(new MaterialPickerOnPositiveButtonClickListener<Long>() {
+            @Override
+            public void onPositiveButtonClick(Long selection) {
+                Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("GMT"));
+                calendar.setTimeInMillis(selection);
+                SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy");
+                sdf.setTimeZone(TimeZone.getTimeZone("GMT"));
+                String formattedDate = sdf.format(calendar.getTime());
+                edtPickUpDate.setText(formattedDate);
+                try {
+                    // Converts the string date back into a date Object
+                    pickupDate = sdf.parse(formattedDate);
+                } catch (ParseException e) {
+                    Log.e(TAG, "error converting the pickup date into a date object", e);
+                }
+                if (checkValidPickUpDate()) {
+                    resetPickUpDateError();
                     calculateNumberOfDays();
                 }
-            });
+            }
         });
     }
 
+
     private void setReturnDateOnClickListener() {
-        edtReturnDate.setOnClickListener(v -> {
-            MaterialDatePicker.Builder<Long> materialDateBuilder = MaterialDatePicker.Builder.datePicker();
-            materialDateBuilder.setTitleText("PICKUP DATE");
-            MaterialDatePicker<Long> materialDatePicker = materialDateBuilder.build();
-            materialDatePicker.show(getSupportFragmentManager(), "MATERIAL_DATE_PICKER");
-            // positive button == ok button
-            materialDatePicker.addOnPositiveButtonClickListener(new MaterialPickerOnPositiveButtonClickListener<Long>() {
-                @Override
-                public void onPositiveButtonClick(Long selection) {
-                    Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
-                    calendar.setTimeInMillis(selection);
-                    @SuppressLint("SimpleDateFormat") SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy");
-                    String formattedDate = sdf.format(calendar.getTime());
-                    edtReturnDate.setText(formattedDate);
-                    try {
-                        // Converts the string date back into a date Object
-                        returnDate = sdf.parse(formattedDate);
-                    } catch (ParseException e) {
-                        Log.e(TAG, "error converting the return date into a date object", e);
-                    }
+        //prevents keyboard from popping up
+        edtReturnDate.setInputType(InputType.TYPE_NULL);
```

Comment:
Why is this null? Why not [TYPE_DATETIME_VARIATION_DATE](https://developer.android.com/reference/android/text/InputType#TYPE_DATETIME_VARIATION_DATE)?

Reply:
It is null, to prevent the keyboard from appearing so the only way for a user to select the data is by clicking the calender displayed.